### DISCUSSION TITLE
[scatter_plot]integrate a generic scatter plot with regression line to Superset as official solution

### DISCUSSION THREAD

**Discussion by junlincc** [apache/superset#18501] [Ideas]
THIS IS A REPOST FROM SUPERSET-UI BY @kaihenzler 🙏

### Is your feature request related to a problem? Please describe.

This is a feature request or rather a request for comments to add Scatter Charts to `@superset-ui/plugin-chart-echarts` 

### Describe the solution you'd like

We developed a Scatter Chart plugin based on echarts in a separate repository for the ease of moving fast and to get a working version up and running: https://github.com/w11k/superset-chart-plugins/blob/master/plugins/plugin-chart-scatter/README.md

The package is published on npm and can be tested via 

`npm install @w11k/plugin-chart-scatter`

and is set up as usual by importing it

```js
import EchartsScatterChartPlugin from '@w11k/plugin-chart-scatter';

new EchartsScatterChartPlugin().configure({ key: 'scatter' }).register();
```

I'm opening this issue to gather feedback if some folks are interested to have the Scatter Chart integrated into this repository. I would go ahead a create a PR that integrates it.

### Describe alternatives you've considered

If there is no interest in integrating the new chart type into this repo the issue can be closed.

### Additional context

There are 2 main versions the chart can be used for.

#### 1. Aggregated Data

X, Y and Bubble Size (optional) can be aggregated and grouped and colored accordingly

![aggregated](https://user-images.githubusercontent.com/4253453/130068969-8d795a1d-dd07-4c8a-a1ce-67e78b69a457.png)

#### 2. Raw Data
![raw](https://user-images.githubusercontent.com/4253453/130068976-02b47ed4-2fe1-41f2-b294-733d13be229e.png)

X, Y and Bubble Size (optional) metric can be selected to plot plain datapoints

#### General Settings

- A Regression line (linear, polynominal, logarithmic, exponential) can be drawn 
- Min/Max Size of the bubble is adjustable
- X,Y Axis Labels are Adjustable
- Legend is adjustable

#### Control Panel

Aggregated Options
![aggregated-options](https://user-images.githubusercontent.com/4253453/130069932-3f59a621-e1b2-44ea-bb63-3ce8cbf90ea4.png)

Raw Options
![raw-options](https://user-images.githubusercontent.com/4253453/130069936-8e119a7c-28c3-45e7-a47c-5cafc86403ae.png)

Common Options
![common-options](https://user-images.githubusercontent.com/4253453/130069934-bbdadd18-9670-4dd7-831a-5166f408ff56.png)

**Comment by junlincc** [comment]
@villebro @stephenLYZ @zhaoyongjie ^

**Comment by serenajiang** [comment]
We'd love to see this! One of our users just requested this today, and it's been a recurring ask over the years.

**Comment by Ayoub-BOUCHACHIA** [comment]
Hello @junlincc,

I'm currently working on creating a custom scatterplot, and I'm starting by using this plugin as a base.

In aggregation mode, I want to put a column on the x-axis and a metric on the y-axis.

Is this possible?
